### PR TITLE
Implement code fix for trailing whitespace after xml doc comments

### LINKED ISSUE #795: SA1028 (CodeMustNotContainTrailingWhitespace) code fix does not support documentation comments
This is a problem for the following reasons:

* Documentation comments are already analyzed, so warnings will be reported.
* The default smart indent implementation in Visual Studio for documentation comments includes trailing whitespace on new lines, as opposed to most other cases where virtual space is used to avoid these spaces.

### PR DESCRIPTION
**Pull request by AArnott**
Fix #795

### CODE REVIEW

**Comment by sharwell** [comment]
Can you add a test which covers a documentation comment in the following form?

```csharp
/**
 * <summary>  
 * Some description    
 * </summary>
 */
class Foo { }
```

**Comment by AArnott** [comment]
I suppose. Do folks actually use that syntax? I just tried it and the language service seems to make it rather difficult.

**Comment by AArnott** [comment]
Hmmmm... it seems #821 which you already linked to tracks support for this in general for this rule. It seems best to keep that as a separate pull request IMO.

**Comment by sharwell** [comment]
I'm OK with that.

File: StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs [hunk 1]
```diff
@@ -71,27 +86,55 @@ public sealed override FixAllProvider GetFixAllProvider()
...
+                                    newTriviaContent = string.Concat(
+                                        oldTriviaContent.Substring(0, diagnosticSpanWithinTrivia.Start),
+                                        oldTriviaContent.Substring(diagnosticSpanWithinTrivia.End));
+                                    var newToken = SyntaxFactory.Token(textToken.LeadingTrivia, textToken.Kind(), newTriviaContent, newTriviaContent, textToken.TrailingTrivia);
+                                    newTextTokens = newTextTokens.Replace(textToken, newToken);
+                                    break;
+                                }
+                            }
+
+                            var newNode = xmlNode.Update(newTextTokens);
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs]: What is the difference between `Update` and `WithTextTokens`?

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs]: Answer: The name. `WithTextTokens` is implemented by calling `Update`.

Comment (AArnott) [inline StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1028CodeFixProvider.cs]: And also: I couldn't find the `WithTextTokens` method for the life of me at the time I was coding this. I suspect I was trying before adding the downcast code I guess.

**Pull request merged**